File: 프로그래머스/2/42747. H－Index/H－Index.py
def solution(citations):
    answer = 0
    
    citations.sort() # 정렬
    
    for h_index in range(0, len(citations)+1):
        more_count = 0
        below_count = 0
        
        for element in citations:
            if h_index <= element:
                more_count += 1
            elif h_index >= element:
                below_count += 1
                
        if h_index <= more_count and h_index >= below_count:
            answer = h_index
            
    return answer

File: 프로그래머스/2/42747. H－Index/H－Index.java
import java.util.*;

class Solution {
    public int solution(int[] citations) {
        int hIndex = 0;
        
        // n : 발표한 논문의 수 
        int n = citations.length;
        
        // 정렬 - 논문의 인용 횟수
        Arrays.sort(citations);
        
        // 반복문을 순회하면서 h-index를 찾자.
        while (true) {
            int countMore = 0;
            int countLess = 0;
            
            for (int i = 0; i < n; i++) {
                int temp = citations[i];
                if (temp >= hIndex) countMore++;
                if (temp < hIndex) countLess++;
            }
            
            if(hIndex <= countMore && hIndex >= countLess){
                hIndex++;
            } else {
                return --hIndex;
            }      
        }

    }
}